### PR TITLE
feat(extension-react-tables): merge two menu components into one

### PR DESCRIPTION
**Pull request by ocavue**
### Description

This PR merges two table menu components (`ButtonComponent` and `PopupComponent`) into one big component. This provides more flexibility for the end-users.

This PR also exports `@remirror/extension-react-tables` from `@remirror/react`. This means that users don't need to install `@remirror/extension-react-tables` manually. (cc @ronnyroeller )

### Checklist

- [x] I have read the [**contributing**](https://github.com/remirror/remirror/blob/HEAD/docs/contributing.md) document.
- [x] My code follows the code style of this project and `pnpm fix` completed successfully.
- [x] I have updated the documentation where necessary.
- [ ] New code is unit tested and all current tests pass when running `pnpm test`.

### CODE REVIEW

File: .changeset/warm-pans-march.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@remirror/extension-react-tables': minor
+---
+
+- Use the new type `TableCellMenuComponentProps` to replace `TableCellMenuComponentProps` and `TableCellMenuProps`.
+- `TableCellMenu` only accept one property `Component` now, which replaces the origin propertys `ButtonComponent` and `PopupComponent`.
```
Comment (ocavue) [inline .changeset/warm-pans-march.md:6]: accept => accepts

**Comment by ocavue** [comment]
@ronnyroeller This PR is necessary to fix the "table menu overflow" issue in Materail-UI. You can check https://github.com/ronnyroeller/cra-remirror-tables/pull/1 for more details. If you think this PR is ok, I could merge it into the beta branch and release a new version.

**Comment by ronnyroeller** [comment]
> This PR is necessary to fix the "table menu overflow" issue in Materail-UI. You can check ronnyroeller/cra-remirror-tables#1 for more details. If you think this PR is ok, I could merge it into the beta branch and release a new version.

Thanks for looking into this. Sorry, I wasn't really clear what I meant. I'm aware that I could use material-ui (or another custom library) to add a solution, i.e. this isn't really an issue for my own usage.
What I meant was: if somebody uses the default implementation of the table menu popup (as in the current Remirror advanced table): they will run exactly in this "table menu overflow" issue.
But maybe this will be considerable easier once https://github.com/remirror/remirror/pull/905 is released...

**Comment by ronnyroeller** [comment]
> If you think this PR is ok, I could merge it into the beta branch and release a new version.

LGTM

**Comment by ocavue** [comment]
@ronnyroeller Yes, you are correct. The default menu implementation in `@remirror/extension-react-tables` still doesn't work. I asked Ifiok and he said that #905 does help this kind of issue, so I will continue to works on this issue once #906 is merged.

File: packages/remirror__react/package.json [hunk 2]
```diff
@@ -43,6 +43,7 @@
     "@remirror/extension-positioner": "1.0.0-next.60",
     "@remirror/extension-react-component": "1.0.0-next.60",
     "@remirror/extension-react-ssr": "1.0.0-next.60",
+    "@remirror/extension-react-tables": "pr706",
```
Comment (ocavue) [inline packages/remirror__react/package.json:46]: Pnpm don't allow me to set the version to `0.0.0`

**Review by ifiokjr — APPROVED** [review]

**Pull request merged**